spec/datadog/tracing/integration_spec.rb

```diff
@@ -24,9 +24,31 @@
   shared_context 'agent-based test' do
     before do
       skip unless ENV['TEST_DATADOG_INTEGRATION']
+
+      # Ensure background Writer worker doesn't wait, making tests faster.
+      stub_const('Datadog::Tracing::Workers::AsyncTransport::DEFAULT_FLUSH_INTERVAL', 0)
+
+      # Capture trace segments as they are about to be serialized
+      segments = trace_segments
+      allow_any_instance_of(Datadog::Transport::TraceFormatter)
+        .to receive(:format!).and_wrap_original do |original|
+          segments << original.call
+        end
     end
 
-    let(:tracer) { Datadog::Tracing.send(:tracer) }
+    def tracer
+      # Do not cache tracer object in a `let`, as trace is recreated on `Datadog.configure`
+      Datadog::Tracing.send(:tracer)
+    end
+
+    let(:trace_segments) { [] }
+    let(:span) do
+      expect(trace_segments).to have(1).item
+      expect(trace_segments[0].spans).to have(1).item
+
+      trace_segments[0].spans[0]
+    end
+    let(:sampling_priority) { span.get_tag('_sampling_priority_v1') }
```

Comment:
Minor: If I'm looking at this correctly, I think this can be moved closer to the tests that actually need it. I think it's especially important to try to do that in a file like `tracing/integration_spec.rb` because it naturally will accumulate more and more complexity over time and so anything we can do to make it easier to understand and maintain I think is worth it.